src/main/java/dev/hello/leetcode/binary_search_tree_to_greater_sum_tree/Solution.java

```diff
@@ -2,28 +2,14 @@
 
 import dev.hello.structures.TreeNode;
 
-import java.util.Stack;
-
 class Solution {
-    public TreeNode bstToGst(TreeNode root) {
-
-        Stack<TreeNode> treeNodeStack = new Stack<>();
-        TreeNode node = root;
-        int sum = 0;
 
-        while (node != null || !treeNodeStack.isEmpty()) {
-            while (node != null) {
-                treeNodeStack.push(node);
-                node = node.right;
-            }
-
-            node = treeNodeStack.pop();
-            sum += node.val;
-            node.val = sum;
-
-            node = node.left;
-        }
+    private int sum = 0;
 
+    public TreeNode bstToGst(TreeNode root) {
+        if (root.right != null) bstToGst(root.right);
+        sum = root.val = sum + root.val;
```

Comment:
```java
root.val = sum + root.val;
sum = root.val;
```

Reply:
Runtime: 0 ms, faster than 100.00% of Java online submissions for Binary Search Tree to Greater Sum Tree.
Memory Usage: 34.1 MB, less than 100.00% of Java online submissions for Binary Search Tree to Greater Sum Tree.